Pipes-Kimberley-Zell-Solar-System.rb

```diff
@@ -0,0 +1,158 @@
+
+# Primary Requirements
+#
+# Create an interface where the user can interact with the solar system and be able to select a planet and view information about it.
+# Allow your user to add their own planet.
+#
+
+
+class Planet
+  attr_accessor :name, :distance, :age, :inhabitants, :color
+
+  def initialize(name,distance,age,inhabitants,color)
+    @name = name
+    @distance = distance
+    @age = age
+    @inhabitants = inhabitants
+    @color = color
+  end
+
+  def to_string
+    return "#{@name}: \n Distance from the Sun: #{@distance} light years\n Age: #{@age} space years \n Inhabitants: #{@inhabitants}\n Color: #{@color}\n\n"
+  end
+end
+
+class SolarSystem
+  attr_accessor :planets
+  def initialize(planets)
+    @planets = planets
+  end
+
+  def planet_add(planet)
+    @planets.push planet
+    return @planet
+  end
+
+  def planet_list
+    num = 0
+    planet_list = ""
+    @planets.each do |planet|
+      num += 1
+      planet_list += "#{num}. #{planet.to_string}\n"
+    end
+    return planet_list
+  end
+
+  def planet_names
+    num = 0
+    planet_names = ""
+    @planets.each do |planet|
+      num += 1
+      planet_names += "#{num}. #{planet.name}\n"
+    end
+    return planet_names
+  end
+
+  # # New Code attempt - does not work ##
+    # def user_adds_planet
+    #   add_planet_hash = {name: nil, age: 0, distance: 0, inhabitants: nil, color: nil}
+    #   return new_planet_name = Planet.new(add_planet_hash[:name], add_planet_hash[:age], add_planet_hash[:distance], add_planet_hash[:inhabitants], add_planet_hash[:color])
+    #   solar_system.planet_add(new_planet_name)
+    #   solar_system.planet_add(new_planet_name)
+    #end
+
+
+
+
+  ## Good code ##
+  def choice(planet_choice)
+    return @planets[planet_choice].to_string
+  end
+
+  #New code attempt#
+  # def choice(planet_choice)
+  #   if planet_choice >= 1 && planet_choice <= solar_system.planet
+  #       return solar_system.planet_names(planet_choice)
+  #   else
+  #     "Invalid answer. Please select a planet number from the planets list."
+  #     planet_choice = gets.chomp.to_i
+  #   end
+  #
+  #   #puts solar_system.choice(planet_choice)
+  # end
+end
+
+# Create Planets using Planet Class
+endor = Planet.new("Endor", 100000, 55000000, "Ewoks", "purple")
+condor = Planet.new("Condor", 125000, 5000, "Condos", "emerald green")
+alderan = Planet.new("Alderan", 92000, 9000000, "Alderians", "white")
+dagobah = Planet.new("Dagobah", 919909, 1883005, "Dagonians", "red")
+jonah = Planet.new("Jonah", 3000452, 1000, "None", "orange")
+solaris = Planet.new("Solaris", 30000, 150, "Solarians", "vermillion")
+# Create array of planets
+planet_array = [endor, condor, alderan, dagobah, jonah, solaris]
```

Comment:
Since you're creating all these local variables just to add them to an array, a more concise way to do this might be:

```ruby
planet_array = [
  Planet.new("Endor", 100000, 55000000, "Ewoks", "purple"),
  Planet.new("Condor", 125000, 5000, "Condos", "emerald green"),
  # ...
]
solar_system = SolarSystem.new(planet_array)
```

This makes it clear to the reader that we won't be using the planets individually, only through the collection.